src/gfn/samplers.py

```diff
@@ -68,7 +67,28 @@ def sample_actions(
                 the sampled actions under the probability distribution of the given
                 states.
         """
-        estimator_output = self.estimator(states)
+        # TODO: Should estimators instead ignore None for the conditioning vector?
```

Comment:
wouldn't it be cleaner with fewer if else blocks ?

Reply:
Ok ! makes sense.

Reply:
I added these `exception_handlers` to reduce the cruft.